sentinel.go

```diff
@@ -0,0 +1,92 @@
+package sentinel
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func New(ctx context.Context, config Config) *sentinel {
+	c := make(chan bool)
+	return &sentinel{
+		Config: config,
+		active: false,
+		ticker: time.NewTicker(config.Duration),
+		stop:   make(chan bool, 1),
+		c:      c,
+		Done:   c,
+	}
+}
+
+func (s *sentinel) Start() error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	if s.active {
+		return errors.New("already started")
+	}
+
+	s.active = true
+	go s.work()
+
+	return nil
+}
+
+func (s *sentinel) Stop() error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	if !s.active {
+		return errors.New("must be active to stop")
+	}
+
+	select {
+	case s.stop <- true:
+		return nil
+	default:
+		return errors.New("stop already requested")
+	}
+}
+
+func (s *sentinel) work() {
+	var manuallyStopped = false
+Loop:
+	for {
+		select {
+		case <-s.stop:
```

Comment:
It might be useful to allow for a manual trigger as well (similar to a manual stop) via a `Trigger()` method